### PR TITLE
Update block ID generation logic

### PR DESCRIPTION
**Pull request by nakulkar-msft**
The block ID names will have a context of the jobID, transfer cookie in the job and block/offset in the source file.
Encoded ID: <5B empty placeholder> <16B GUID of AzCopy re-interpreted as string><5B PartNum><5B Index in the jobPart><5B blockNum>

### CODE REVIEW

**Review by tasherif-msft — COMMENTED** [review]
Please provide a pr description with:
1. why we're doing this change
2. a brief comment on what the change is
Also, can this be tested?

**Comment by nakulkar-msft** [comment]
> Please provide a pr description with:
> 
> 1. why we're doing this change
> 2. a brief comment on what the change is
>    Also, can this be tested?

Shared the reason on email. 
Testing would be check that we are generating same blockID repeatedly for the same input.

**Comment by nakulkar-msft** [comment]
@adreed-msft @tasherif-msft

**Review by tasherif-msft — APPROVED** [review]

File: common/extensions.go [hunk 1]
```diff
@@ -178,3 +182,9 @@ func GenerateFullPathWithQuery(rootPath, childPath, extraQuery string) string {
 		return p + "?" + extraQuery
 	}
 }
+
+func GenerateBlockBlobBlockID(blockNamePrefix string, index int32) string {
```
Comment (tasherif-msft) [inline common/extensions.go:186]: write a small comment above the method mentioning what's encoded in the ID

**Review by vibhansa-msft — APPROVED** [review]

**Pull request merged**